x-pack/plugins/xpack_main/server/lib/xpack_info.js

```diff
@@ -260,4 +260,22 @@ export class XPackInfo {
 
     return newLicense.expiry_date_in_millis !== cachedLicense.expiry_date_in_millis;
   }
+
+  /**
+   * Handles the case where the license type is basic, and the date is so far into the
+   * future that moment cannot parse it. In that case we should treat it as never expiring.
+   * @param {Object} xpackInfo xPack info response object returned from the backend.
+   * @returns {string} 'never' if license is basic and date unparseable by moment;
+   *                   otherwise, a formatted date.
+   * @private
+   */
+  _getExpiryDate(xpackInfo) {
+    let expiryDate = moment(get(xpackInfo, 'license.expiry_date_in_millis')).format('YYYY MMM DD HH:mm');
+
+    if (expiryDate === 'Invalid date' && get(xpackInfo, 'license.mode') === 'basic') {
```

Comment:
`this.license.getType()` doesn't work here. is this the right way to check for basic license??